internal/pkg/orchestrator/orchestrator.go

```diff
@@ -0,0 +1,221 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package orchestrator
+
+import (
+	"context"
+	"strings"
+	"time"
+
+	"go.opentelemetry.io/auto/internal/pkg/errors"
+	"go.opentelemetry.io/auto/internal/pkg/instrumentors"
+	"go.opentelemetry.io/auto/internal/pkg/instrumentors/bpffs"
+	"go.opentelemetry.io/auto/internal/pkg/log"
+	"go.opentelemetry.io/auto/internal/pkg/opentelemetry"
+	"go.opentelemetry.io/auto/internal/pkg/process"
+)
+
+type pidServiceName struct {
+	serviceName string
+	pid         int
+}
+
+// New creates a new Implementation of orchestrator Service.
+func New(
```

Comment:
```suggestion
func NewService(
```

The created type is not named `Orchestrator`.